src/components/storage/ReclaimSpaceModal.jsx

```diff
@@ -359,14 +355,16 @@ const DeviceActionDelete = ({ device, hasBeenRemoved, newDeviceSize, onAction })
 
     // Disable the remove action for disk devices without partitions
     const isRemoveDisabled = device.type.v === "disk" && device.children.v.length === 0;
```

Comment:
Please integrate the second check here:
 const isDiskWithoutChildren = device.type.v === "disk" && device.children.v.length === 0;
 const isExtendedPartition = device.attrs?.v.isleaf === "False" && device.children.v.length === 0;
 const isRemoveDisabled = isDiskWithoutChildren || isExtendedPartition;